### DISCUSSION TITLE
tedana roadmap discussion

### DISCUSSION THREAD

**Discussion by handwerkerd** [ME-ICA/tedana#879] [Ideas]
Several years ago, a bunch of of us put together a [roadmap for tedana](https://tedana.readthedocs.io/en/stable/roadmap.html). We're actually getting close to achieving the main goals of the roadmap! We're definitely not there yet, but this is a good time to start thinking ahead.

We can host this discussion here. As people put forward ideas, here are a few categories to consider:
- Current stuff better: A functionally that tedana currently has, but you have a way you want to see it improve
- Novel methodology: Ways to improve the underlying methods
- Connections to other tools: Are there parts of tedana that should be more accessible to external programs
- Software Documentation
- Multi-echo fMRI education documentation & resources
- Shared datasets for education, validation and testing

**Comment by handwerkerd** [comment]
Starting off with some ideas several of us have been discussing over the years.

- Current stuff better
  - More ways to visualize and interact with logged information from a tedana execution
    - Flow chart visualization of an executed decision tree
    - Tools to collect and organized summary measures across runs and populations (i.e variance explained, number of components, variance in accepted vs rejected) 
- Novel methodology
  - Find better ways to estimate appropriate variance explained for PCA/ICA (maybe going back to using multi-echo information)
  - Switch away form ICA to a method that can maximize the distance between kappa & rho within each component rather than maximizing statistical independence as ICA does
  - As the core measure of what makes a component good/bad are kappa & rho and the elbow calculations the best we can do?
  - Better integrate non-multi-echo noise removal tools with multi-echo noise removal (parts of which might be beyond the scope of tedana, but making connections could be within the scope)
- Connections to other tools
  - Continue work on modularization code to make it easier to have other programs connect to parts
  - Reword the main tedana.py workflow so that it's easier to access parts and easier for contributors to improve parts without needs to understand how everything fits together
  - Decide of other parts of tedana should become stand-alone packages (maybe the modularized decision tree and metric calculation sections???
- Multi-echo fMRI education documentation & resources
  - Expand the multi-echo analysis Jupyter book:https://me-ica.github.io/multi-echo-data-analysis
- Shared datasets for education, validation and testing
  - Build up validation tools to support comparisons across multiple data sets and decision trees.
  - Decide how we can make sharing data specifically for tedana development & validation easier.

**Comment by eurunuela** [comment]
@dowdlelt and I brainstormed about the novel methodology that could be added to or supported by tedana. The list of ideas is too unstructured and unpolished to post here (can be accessed [here](https://docs.google.com/document/d/1obMIE8PBy7dsWB57eLj-9MvSSi-oY7Yqxgh25tI7_bE/edit?usp=sharing)), but I guess we will edit this comment as we write them more clearly.

To improve classification:
  - Compare component time series to motion regressors
  - Compare time series with physiological regressors
  - Examine data for slice-to-slice artifacts (spatial FFT over slice dimension - high-frequency component means movement artifact probably ( high frequency, in this case, being slice to slice)
  - Provide slice timing (or JSON) Multiband artifact - are spatial maps of simultaneous slices more similar than actual neighbor slices. Probably Multiband issue, noise
  - Symmetrical (left-right) smooth activation (or symmetrical absolute values) should probably be classified as bold-like
  - Tissue maps, does it look like an edge effect? Csf? Etc
  - Reconstruction time series from unclear components, smooth data, and try again - maybe bold components would be more clear and could be classified
  - Provide Field strength - how much does a component overlap with CSF like t2star values, wm t2star values, etc
  - Provide Field map, how much does component overlap with high susceptibility regions (identify respirations maybe)
  - Provide task if available - try and keep components that look like a task ( controversial)
  - Provide Field strength - convert components to percent signal change - if 3t and percent signal change reaches 20, that's noise! (Probably, requires testing)
  - Extracts skull from the first echo, if components overlap with that, it's noise!
  - Revisiting the smoothing idea - kappa values should remain stable as smoothing increases, perhaps there is a kappa+1 voxel smoothing, kappa+2 voxel smoothing, etc columns
- Multiple runs, cross-validation (hard)
- Multiple runs, concatenation - MELODIC takes multiple runs for ICA, tedana should too. Increase sensitivity! More components to sort, but BOLD should be similar across runs ( and reinforce) whereas noise should vary from run to run
- Actually, there is an idea, that even if you tedana each run independently, you could add another column, does a component have a match in another run. Consider spatial - networks (or task activation) should be similar run to run so one component would be very very similar to one of the components in another run.
  - Movement could be similar, but probably not
  - Physiological would be similar but could be identified through other methods.

Metrics:
- Currently, kappa is based on some math done to the spatial map. What if we instead took a grey matter mask and calculated, within that only. And within WM only. A kappa grey to white ratio. It's another metric
- And can we calculate a kappa-like value based not on the spatial maps of each echo, but on the kappa/rho nature of each component's timecourse?

Rica:
- The SVM (or whatever, could be deep learning too) trained on Rica classifications idea for "automagic" classification of components.
  - Rica would upload the classification file to a server, and these manual classifications would be used to train a classification algorithm.
